common/evidence.py

```diff
@@ -238,23 +280,25 @@ def read_path(path: str, spark_instance) -> DataFrame:
         return spark_instance.read.option("recursiveFileLookup", "true").json(path)
 
     # A directory with Parquet files.
-    if parquet_files:
+    else:
```

Comment:
This wouldn't work if the folder contained CSVs.

Reply:
If the path only contains csv-s, the above assertion would have already failed:

```python
assert (
    json_files or parquet_files
), f"The provided directory {path} contains neither JSON nor Parquet."
```

However if a folder contains a mix of csv and parquet file it would indeed fail, but not because of the condition, but because of this in line 284:

```python
        return spark_instance.read.parquet(path)
```

More over this would also fail:

```python
    if json_files:
        return spark_instance.read.option("recursiveFileLookup", "true").json(path)
```

I think it is safe to assume if a directory is provided, it is either a propery partitioned json or parquet dataset. If it's a partitioned tsv, it won't fly.